src/Akka.Persistence.Sql/Query/SqlReadJournal.cs

```diff
@@ -219,13 +219,14 @@ private Source<EventEnvelope, NotUsed> EventsByPersistenceIdSource(
                             timestamp: r.representation.Timestamp, 
                             tags: r.tags));
 
-        private Source<EventEnvelope, NotUsed> CurrentJournalEvents(long offset, long max, MaxOrderingId latestOrdering)
+        private Source<EventEnvelope, Task<long>> CurrentJournalEvents(long offset, long max, MaxOrderingId latestOrdering)
         {
             if (latestOrdering.Max < offset)
-                return Source.Empty<EventEnvelope>();
+                return Source.Empty<EventEnvelope>().MapMaterializedValue(_ => Task.FromResult(0L));
 
             return _readJournalDao
                 .Events(offset, latestOrdering.Max, max)
+                .AlsoToMaterialized(Sink.Aggregate<Try<(IPersistentRepresentation, string[], long)>, long>(0L, (readCount, _) => readCount + 1), Keep.Right)
```

Comment:
.... I can't help but ask if we should just enrich `Events` return flow here instead, relying on this aggregate doesn't look easy to comprehend/maintain in my head.

CC @Arkatufus + @Aaronontheweb , if we don't ever need equality on the value of these `Try` bits, it can be better from AOT perspective to just declare our own struct since then it doesn't have to generate generic equality code for the type.